server/jetstream.go

```diff
@@ -206,13 +216,24 @@ func (s *Server) checkStoreDir(cfg *JetStreamConfig) {
 			// If this seems to be an account go ahead and move the directory. This will include all assets
 			// like streams and consumers.
 			if ok {
+				if !haveJetstreamDir {
```

Comment:
With the spirit of bailing out on any error, which I don't disagree with, then what about the lookup error few lines above? We are currently simply logging but move on. Should we bail out?
My only concern is that is the account resolver always known to be fully setup at this point in the startup sequence? Are we going to get to a point with some configs where the server won't be able to restart?

Reply:
There's a point at which "starting up without data" is far more dangerous than "refusing to start".  In this case, we risk creating a divergent forked view of what state exists which is hard to reconcile, and the safe option is to refuse to start and require administrator intervention, instead of effectively continuing on without precious state.

User-data in jetstream is precious state.  If we can't handle the migration, we should not start.

The main cause of errors is filesystem errors.  Primarily because of detritus from 2.2.3 which is still not handled.  The fixes here improve things to where we will successfully migrate from 2.2.2 to 2.2.4 and where we will not silently fail if 2.2.3 layout exists.  It's better.  It's not perfect.

Reply:
My review also requested specifically checking the assumption that a `jetstream` dir actually exist rather than trusting len()==1 etc.

Reply:
My point was that: should we go a step further and return error when account lookup fails?

Reply:
Sorry, I missed the details of your question @kozlovic. 

I don't know what the right answer is about the account lookup - but this was also in my initial review since account servers are not reliable or to be trusted.  At the very least we should log loudly in detail and we should ensure that on the next startup we will again try to migrate until nothing is left.

Problem is we have no way of knowing what nothing is left mean since we do not have a clear enumerable way to determine that so I really don't know what the answer is. If migration is a one-shot attempt because we are not capable of picking this up reliably then we need to log loudly manual instructions for moving this data.

Definitely coming around to @philpennock's suggestion that we should have pulled 2.2.3.

Reply:
Re: pulling 2.2.3.. Not sure we can remove from Docker hub official images.. which is likely where most people get the release from.

Reply:
If 2.2.4 is imminent then perhaps no longer worth it this time, but perhaps a docker-library/official-images PR which moves `latest` back to a known-good release is the right way to go, even if we can't actually remove a tag.

Reply:
Here are going to remove "now", so hopefully no need to do that.